mailet/standard/src/test/java/org/apache/james/transport/mailets/AddFooterTest.java

```diff
@@ -35,34 +35,25 @@
 import org.apache.mailet.base.test.FakeMail;
 import org.apache.mailet.base.test.FakeMailetConfig;
 import org.apache.mailet.base.test.MailUtil;
-import org.junit.Before;
-import org.junit.Rule;
-import org.junit.Test;
-import org.junit.rules.ExpectedException;
-import org.junit.runner.RunWith;
-import org.junit.runners.Parameterized;
-import org.junit.runners.Parameterized.Parameters;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
 
 import com.google.common.base.Joiner;
-import com.google.common.collect.ImmutableList;
 import com.google.common.collect.Lists;
-import com.google.common.collect.Sets;
 
-@RunWith(Parameterized.class)
-public class AddFooterTest {
+class AddFooterTest {
 
     private static final String MY_FOOTER = "my footer";
-    @Rule public ExpectedException exception = ExpectedException.none();
-    
-    private final String javaCharset;
-    private final String javaMailCharset;
+
     private Mailet mailet;
     
-    @Parameters
-    public static Collection<String[]> data() {
+
+    private static Stream<Arguments> charsetTuples() {
         //javamail has its own charset handling logic, it needs to be exercised
-        List<String[]> charsets = Lists.newArrayList();
-        Set<String> charsetNamesToTest = Sets.newHashSet(
+        List<String> charsetNamesToTest = Lists.newArrayList(
```

Comment:
you can use directly Stream.of() instead of new list then .stream(), no?

Reply:
no, because you need to consume it twice